packages/twenty-server/src/engine/metadata-modules/serverless-function/serverless-function.service.ts

```diff
@@ -60,13 +91,73 @@ export class ServerlessFunctionService extends TypeOrmQueryService<ServerlessFun
     if (
       functionToExecute.syncStatus === ServerlessFunctionSyncStatus.NOT_READY
     ) {
+      await this.serverlessService.build(functionToExecute, version);
+      await super.updateOne(functionToExecute.id, {
+        syncStatus: ServerlessFunctionSyncStatus.READY,
+      });
+    }
+
+    return this.serverlessService.execute(functionToExecute, payload, version);
+  }
+
+  async publishOneServerlessFunction(id: string, workspaceId: string) {
```

Comment:
should we also delete the previous version? There is no way for a user to access it today. If the design change so we can reset to an older version we would remove it. But for now it would avoid keeping useless versions. Wdyt?

Reply:
I think we should keep the previous versions as it is the whole point of that feature.
It will be useful if we need that workflows steps be able to run different versions of the serverless function for example.
Through the graphql API it is possible to get the previous code versions, and to execute previous versions. 
I think we can leave it like this as it will not impact our server ressources, and it will simplify the development of potential future version management which I am sure will arrive pretty soon.